### ISSUE TITLE
Frozen dataclasses should only prevent mutation of instance variables

### ISSUE THREAD

**Issue by vberlier** [microsoft/pyright#1483]
Pyright reports an error when initializing class variables for frozen dataclasses:

```python
@dataclass(frozen=True)
class Point:
    x: float
    y: float

    ZERO: ClassVar["Point"]

Point.ZERO = Point(0, 0)  # Cannot assign member "ZERO" for type "Type[Point]"
                          #   Dataclass "Point" is frozen
```

`frozen=True` should only affect instance variables.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.110, which I just published. It will also be included in the next release of Pylance.
